packages/jellyfish-wallet/src/wallet_account.ts

```diff
@@ -45,10 +45,14 @@ export abstract class WalletAccount implements WalletEllipticPair {
    *
    * @param {string} address to parse into script
    * @return {Script} parsed from address
+   * @throws {Error} if address or network is invalid
    */
   addressToScript (address: string): Script {
-    const parsed = DeFiAddress.from(this.network.name, address)
-    return parsed.getScript()
+    const decodedAddress = fromAddress(address, this.network.name)
+    if (decodedAddress?.script === undefined) {
```

Comment:
This can possibly be simplified.
```suggestion
    if (!decodedAddress?.script) {
```
I believe from implementation, it can only be `undefined` or `DecodedAddress`, thus there should be no issue with falsy error when comparing.

Another more explicit way.
```suggestion
    if (decodedAddress? .script) {
        return decodedAddress.script
    }
    throw new Error('address is invalid')
```
Just preferences, not a must.

Reply:
Actually, this is enforced by the linting engine, explicit over implicit, and no implicit coercion. Because this synthetic sugar provides somewhat of a greater DX - it can introduce ambiguity and unnecessary confusion in a large codebase.

Reply:
Also on the topic of why I prefer returning errors early over returning objects. It's another explicit DX where you move explicit errors up to terminate early while leaving one true condition that it can allow the function to complete. 

This makes the code base more manageable and maintainable as you allow more injection of errors checking through adding more early termination. See this from a code complexity standpoint and long-term maintainability.

Reply:
Ah got it. Thanks for the explaination.